tests/test_oic_message.py

```diff
@@ -953,3 +960,127 @@ def test_verify_token_encrypted_no_key():
+
+        assert lt.verify()
+
+    def test_with_sub_and_sid(self):
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={BACK_CHANNEL_LOGOUT_EVENT: {}},
+                         iat=utc_time_sans_frac(), jti=rndstr(16),
+                         sub='https://example.com/sub', sid=rndstr())
+
+        assert lt.verify()
+
+    def test_no_sub_or_sid(self):
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={BACK_CHANNEL_LOGOUT_EVENT: {}},
+                         iat=utc_time_sans_frac(), jti=rndstr(16))
+
+        with pytest.raises(ValueError):
+            lt.verify()
+
+    def test_extra_event(self):
+        # more the one event
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={
+                             BACK_CHANNEL_LOGOUT_EVENT: {},
+                             'http://schemas.openid.net/event/other}': {}
+                         },
+                         jti=rndstr(16),
+                         iat=utc_time_sans_frac(), sub='https://example.com/sub')
+
+        with pytest.raises(ValueError):
+            lt.verify()
+
+    def test_wrong_event(self):
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={
+                             'http://schemas.openid.net/event/other}': {}
+                         },
+                         jti=rndstr(16),
+                         iat=utc_time_sans_frac(), sub='https://example.com/sub')
+
+        with pytest.raises(ValueError):
+            lt.verify()
+
+    def test_wrong_aud(self):
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={BACK_CHANNEL_LOGOUT_EVENT: {}},
+                         iat=utc_time_sans_frac(), jti=rndstr(16),
+                         sub='https://example.com/sub')
+
+        with pytest.raises(NotForMe):
+            lt.verify(aud='https://example.com')
+
+    def test_wrong_iss(self):
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={BACK_CHANNEL_LOGOUT_EVENT: {}},
+                         iat=utc_time_sans_frac(), jti=rndstr(16),
+                         sub='https://example.com/sub')
+
+        with pytest.raises(NotForMe):
+            lt.verify(iss='https://rp.example.org')
+
+    def test_wrong_iat(self):
+        # Issued sometime in the future
+        lt = LogoutToken(iss='https://example.com', aud=['https://rp.example.org'],
+                         events={BACK_CHANNEL_LOGOUT_EVENT: {}},
+                         iat=utc_time_sans_frac()+86400, jti=rndstr(16),
+                         sub='https://example.com/sub')
+
+        with pytest.raises(ValueError):
+            lt.verify()
+
+
+class TestBackchannelLogout(object):
```

Comment:
The test for ID token verification failure is missing.

Reply:
You mean Logout Token ?
There is no ID Token involved.

Reply:
Yes, sorry. I meant LogoutToken. A test where LogoutToken fails to verify is still missing.